src/surfaceintersection.jl

```diff
@@ -0,0 +1,124 @@
+# Detect if the solution crossed a root of event function g
+surfacecrossing(g_old::Taylor1{T}, g::Taylor1{T}, eventorder::Int) where {T <: Real} = g_old[eventorder+1]*g[eventorder+1] < zero(T)
+surfacecrossing(g_old::Taylor1{Taylor1{T}}, g::Taylor1{Taylor1{T}}, eventorder::Int) where {T <: Real} = g_old[eventorder+1][1]*g[eventorder+1][1] < zero(T)
+surfacecrossing(g_old::Taylor1{TaylorN{T}}, g::Taylor1{TaylorN{T}}, eventorder::Int) where {T <: Real} = g_old[eventorder+1][1][1]*g[eventorder+1][1][1] < zero(T)
+
+# An rudimentary convergence criterion for the Newton-Raphson root-finding process
+nrconvergencecriterion(g_val::T, nrabstol::T, nriter::Int, maxnriters::Int) where {T<:Real} = abs(g_val) > nrabstol && nriter ≤ maxnriters
+nrconvergencecriterion(g_val::Taylor1{T}, nrabstol::T, nriter::Int, maxnriters::Int) where {T<:Real} = abs(g_val[1]) > nrabstol && nriter ≤ maxnriters
+nrconvergencecriterion(g_val::TaylorN{T}, nrabstol::T, nriter::Int, maxnriters::Int) where {T<:Real} = abs(g_val[1][1]) > nrabstol && nriter ≤ maxnriters
+
+# This function should be moved to TaylorSeries.jl, but the definitive implementation still should be discussed
+function deriv(n::Int, a::Taylor1{T}) where {T <: Number}
+    @assert a.order ≥ n ≥ 0
+    if n==0
+        return a
+    elseif n==1
+        return derivative(a)
+    else
+        return deriv(n-1, derivative(a))
+    end
+end
+
+function taylorinteg(f!, g, q0::Array{U,1}, t0::T, tmax::T,
```

Comment:
Add docstrings here to explain what is `g` and how to use it.

Reply:
Sure, I will add them, thanks!